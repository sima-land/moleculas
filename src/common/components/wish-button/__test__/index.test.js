import React from 'react';
import { shallow, mount } from 'enzyme';
import WishButton from '../index';

describe('<WishButton />', () => {
  it('should render without props', () => {
    const wrapper = mount(<WishButton />);
    expect(wrapper).toMatchSnapshot();
  });
  it('should render with props', () => {
    const onClick = jest.fn();
    const wrapper = mount(<WishButton />);
    expect(wrapper).toMatchSnapshot();
    wrapper.setProps({ isWished: true, onClick });
    expect(wrapper).toMatchSnapshot();
  });
  it('onClick not to have been called if isFetchingWishItems', () => {
    const spy = jest.fn();
    const wrapper = shallow(
      <WishButton
        onClick={spy}
        isFetchingWishItems
      />
    );
    expect(spy).toHaveBeenCalledTimes(0);

    wrapper.simulate('click');
    expect(spy).toHaveBeenCalledTimes(0);

    wrapper.setProps({ isFetchingWishItems: false });
    wrapper.simulate('click');
    expect(spy).toHaveBeenCalledTimes(1);
  });
});
