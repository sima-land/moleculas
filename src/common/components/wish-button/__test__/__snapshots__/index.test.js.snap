// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<WishButton /> should render with props 1`] = `
<WishButton
  isWished={true}
  onClick={[MockFunction]}
>
  <div
    className="wish-button is-wished"
    onClick={[MockFunction]}
    title="Убрать из избранного"
  >
    <Icon
      className="stroke"
      icon={[Function]}
      size={24}
      viewBox="0 0 24 24"
    >
      <heartFilledIcon
        className="icon icon-block stroke"
        height={24}
        viewBox="0 0 24 24"
        width={24}
        xmlns="http://www.w3.org/2000/svg"
      >
        <svg
          className="icon icon-block stroke"
          height={24}
          viewBox="0 0 24 24"
          width={24}
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M19.657 4.01C17.018 2.174 13.76 3.031 12 5.132c-1.759-2.1-5.018-2.968-7.657-1.122-1.4.98-2.279 2.632-2.339 4.376-.14 3.958 3.299 7.13 8.547 11.996l.1.091c.76.704 1.93.704 2.689-.01l.11-.102c5.248-4.855 8.676-8.027 8.546-11.985-.06-1.734-.94-3.386-2.339-4.366z"
          />
        </svg>
      </heartFilledIcon>
    </Icon>
  </div>
</WishButton>
`;

exports[`<WishButton /> should render without props 1`] = `
<WishButton>
  <div
    className="wish-button"
    title="Добавить в избранное"
  >
    <Icon
      className="stroke"
      icon={[Function]}
      size={24}
      viewBox="0 0 24 24"
    >
      <heartIcon
        className="icon icon-block stroke"
        height={24}
        viewBox="0 0 24 24"
        width={24}
        xmlns="http://www.w3.org/2000/svg"
      >
        <svg
          className="icon icon-block stroke"
          height={24}
          viewBox="0 0 24 24"
          width={24}
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M19.944 3.6h-.001c-2.7-1.879-5.978-1.165-7.943.793-1.964-1.958-5.241-2.683-7.943-.792C2.524 4.673 1.57 6.473 1.504 8.369c-.075 2.145.826 4.042 2.36 5.985 1.512 1.912 3.685 3.927 6.261 6.315l.086.08h.002l.098.091a2.466 2.466 0 0 0 3.371-.012l.107-.1.027-.025c2.602-2.407 4.795-4.436 6.316-6.361 1.534-1.941 2.435-3.838 2.364-5.982v-.001c-.065-1.886-1.02-3.686-2.552-4.758zM12 18.274c-2.278-2.105-4.14-3.84-5.45-5.401-1.358-1.62-2.047-2.968-2.047-4.261 0-1.774 1.285-3.07 2.999-3.07 1.055 0 2.103.553 2.72 1.39.38.516.996 1.017 1.78 1.017.789 0 1.406-.507 1.787-1.025.612-.831 1.657-1.382 2.71-1.382 1.713 0 2.998 1.296 2.998 3.07 0 1.293-.689 2.641-2.047 4.26-1.309 1.561-3.171 3.296-5.45 5.402z"
          />
        </svg>
      </heartIcon>
    </Icon>
  </div>
</WishButton>
`;
